plasmapy/particles/ionization_state.py

```diff
@@ -307,6 +308,9 @@ def __init__(
                 )
 
         self._particle = particle
+        self._particle_list = ParticleList(
+            [Particle(particle, Z=i) for i in range(0, particle.atomic_number + 1)]
```

Comment:
```suggestion
            [Particle(particle, Z=i) for i in range(0, self._number_of_particles)]
```

This got added recently. No reason not to use it! :)

Also, I kind of wonder whether you might not be able to use a generator here, just like the first of the following:

```python
[ins] In [1]: list(x**2 for x in range(10))
Out[1]: [0, 1, 4, 9, 16, 25, 36, 49, 64, 81]

[ins] In [2]: list([x**2 for x in range(10)])
Out[2]: [0, 1, 4, 9, 16, 25, 36, 49, 64, 81]
````

But there doesn't seem to be any actual performance difference between the two, so, meh.

Reply:
Oh good point!  I like the idea for using a generator here.

And now that you mention it, I do remember making a code review comment on that very attribute!